codex-rs/network-proxy/src/config.rs

```diff
@@ -0,0 +1,203 @@
+use anyhow::Context;
+use anyhow::Result;
+use codex_core::config::default_config_path;
+use serde::Deserialize;
+use serde::Serialize;
+use std::net::IpAddr;
+use std::net::SocketAddr;
+use std::path::PathBuf;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Config {
+    #[serde(default)]
+    pub network_proxy: NetworkProxyConfig,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            network_proxy: NetworkProxyConfig::default(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NetworkProxyConfig {
+    #[serde(default)]
+    pub enabled: bool,
+    #[serde(default = "default_proxy_url")]
+    pub proxy_url: String,
+    #[serde(default = "default_admin_url")]
+    pub admin_url: String,
+    #[serde(default)]
+    pub mode: NetworkMode,
+    #[serde(default)]
+    pub policy: NetworkPolicy,
+    #[serde(default)]
+    pub mitm: MitmConfig,
+}
+
+impl Default for NetworkProxyConfig {
+    fn default() -> Self {
+        Self {
+            enabled: false,
+            proxy_url: default_proxy_url(),
+            admin_url: default_admin_url(),
+            mode: NetworkMode::default(),
+            policy: NetworkPolicy::default(),
+            mitm: MitmConfig::default(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NetworkPolicy {
+    #[serde(default, rename = "allowed_domains", alias = "allowedDomains")]
```

Comment:
instead of an individual `alias` for each, add `#[serde(rename_all = "camelCase")]` to the `struct` declaration.

Is there any reason we should support two names for each field?